ui/src/controller/track_filter.ts

```diff
@@ -0,0 +1,198 @@
+// Copyright (C) 2023 The Android Open Source Project
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { Registry } from "../common/registry";
+import { AddTrackArgs, AddTrackGroupArgs } from "../common/actions";
+import { globals } from "../frontend";
+import { assertTrue } from "../base/logging";
+
+export const TRACK_GROUP_KIND = '__track_group__';
+
+export type FilterAction = 'include' | 'exclude' | 'pass';
```

Comment:
For my understanding. We need the `pass` option, because we only want to show tracks that are included by at least one filter and not excluded by any filter? So if all filter pass, the track will not be shown. Is this correct?